### DISCUSSION TITLE
How do I add custom LLM APIs to the Langfuse UI?

### DISCUSSION THREAD

**Discussion by yudhiesh** [langfuse/langfuse#1353] [Support]
I am trying to see how I can integrate LLMs that I self-host into the Langfuse UI for prompt engineering tracking. I am aware that we can use the model config, but how do I extend it to support my custom APIs?

**Comment by marcklingen** [comment]
How is the model/LLM-API coupled to the prompt? Generally you can version the prompt in Langfuse, pull it in your application, and then use it with whatever model you want. If your model has any parameters that you'd like to edit as well in Langfuse, you can add them to the model config.
